packages/@react-aria/overlays/src/usePreventScroll.ts

```diff
@@ -204,16 +215,25 @@ function preventScrollMobileSafari() {
     // Restore styles and scroll the page back to where it was.
     restoreStyles();
     removeEvents();
-    window.scrollTo(scrollX, scrollY);
+    if (preventScrollCount === 1) {
+      window.scrollTo(originalValues.get('pageXOffset') as number, originalValues.get('pageYOffset') as number);
+    }
+    preventScrollCount--;
```

Comment:
I like doing it in this order better, easier to read that we're looking to revert if the count drops to zero
```suggestion
    preventScrollCount--;
    if (preventScrollCount <= 0) {
      window.scrollTo(originalValues.get('pageXOffset') as number, originalValues.get('pageYOffset') as number);
    }
```

Reply:
lets also reset `originalValues` so it can't sit around with old values in it